src/sql/workbench/contrib/notebook/test/electron-browser/notebookFindModel.test.ts

```diff
@@ -0,0 +1,172 @@
+		cell_type: CellTypes.Code,
+		source: 'insert into t1 values (c1, c2) \ninsert into markdown values (*hello worls*)',
+		metadata: { language: 'python' },
+		execution_count: 1
+	}, {
+		cell_type: CellTypes.Markdown,
+		source: 'I am *markdown*',
+		metadata: { language: 'python' },
+		execution_count: 1
+	}],
+	metadata: {
+		kernelspec: {
+			name: 'mssql',
+			language: 'sql'
+		}
+	},
+	nbformat: 4,
+	nbformat_minor: 5
+};
+
+let defaultUri = URI.file('/some/path.ipynb');
+
+let mockClientSession: TypeMoq.Mock<IClientSession>;
+let sessionReady: Deferred<void>;
+let mockModelFactory: TypeMoq.Mock<ModelFactory>;
+let notificationService: TypeMoq.Mock<INotificationService>;
+let capabilitiesService: TypeMoq.Mock<ICapabilitiesService>;
+let instantiationService: IInstantiationService;
+let serviceCollection = new ServiceCollection();
+
+suite('Notebook Find Model', function (): void {
+
+	let notebookManagers = [new NotebookManagerStub()];
+	let memento: TypeMoq.Mock<Memento>;
+	let queryConnectionService: TypeMoq.Mock<TestConnectionManagementService>;
+	let defaultModelOptions: INotebookModelOptions;
+	const logService = new NullLogService();
+	let model: NotebookModel;
+	//const retryCount = 24; // 2 minutes
+
+	setup(async () => {
+		sessionReady = new Deferred<void>();
+		notificationService = TypeMoq.Mock.ofType(TestNotificationService, TypeMoq.MockBehavior.Loose);
+		capabilitiesService = TypeMoq.Mock.ofType(TestCapabilitiesService);
+		memento = TypeMoq.Mock.ofType(Memento, TypeMoq.MockBehavior.Loose, '');
+		memento.setup(x => x.getMemento(TypeMoq.It.isAny())).returns(() => void 0);
+		queryConnectionService = TypeMoq.Mock.ofType(TestConnectionManagementService, TypeMoq.MockBehavior.Loose, memento.object, undefined, new TestStorageService());
+		queryConnectionService.callBase = true;
+
+		instantiationService = new InstantiationService(serviceCollection, true);
+		defaultModelOptions = {
+			notebookUri: defaultUri,
+			factory: new ModelFactory(instantiationService),
+			notebookManagers,
+			contentManager: undefined,
+			notificationService: notificationService.object,
+			connectionService: queryConnectionService.object,
+			providerId: 'SQL',
+			cellMagicMapper: undefined,
+			defaultKernel: undefined,
+			layoutChanged: undefined,
+			capabilitiesService: capabilitiesService.object
+		};
+		mockClientSession = TypeMoq.Mock.ofType(ClientSession, undefined, defaultModelOptions);
+		mockClientSession.setup(c => c.initialize()).returns(() => {
+			return Promise.resolve();
+		});
+		mockClientSession.setup(c => c.ready).returns(() => sessionReady.promise);
+		mockModelFactory = TypeMoq.Mock.ofType(ModelFactory);
+		mockModelFactory.callBase = true;
+		mockModelFactory.setup(f => f.createClientSession(TypeMoq.It.isAny())).returns(() => {
+			return mockClientSession.object;
+		});
+
+		await initNotebookModel(expectedNotebookContent);
+	});
+
+	test('Should find results in the notebook', async function (): Promise<void> {
```

Comment:
Did you check the code coverage results to make sure you're covering all the lines? It doesn't look like you're testing the scenario where the cellVal is an array of strings.